endpoints/main.go

```diff
@@ -28,17 +31,25 @@ var totalFailedUptimeChecksCounterVec *prometheus.CounterVec
 // Record the metrics into the appropriate prometheus register under the label specified by organizationName
 // recordLongRunningMetrics specifies wether or not to record information contained in the capability statment
 func getHTTPRequestTiming(urlString string, organizationName string, recordLongRunningMetrics bool) {
-	var resp, responeTime, err = querier.GetResponseAndTiming(urlString)
+	var resp, responeTime, cancelFunc, err = querier.GetResponseAndTiming(urlString)
+
+	if cancelFunc != nil {
+		// Cancel the context once we are done with this function so that context does not remain in memory (causing a leak)
+		// We need to call cancelFunc in this context because we are still reading from the response Body in this context
+		defer cancelFunc()
+	}
 
 	if err != nil {
+		log.WithFields(log.Fields{"organization": organizationName, "url": urlString}).Warn("Error sending getting response charactaristics for endpoint.", err.Error())
```

Comment:
Should this message be "Error **getting** response charactaristics for endpoint." or "Error **sending** response charactaristics for endpoint."? Looks like there's a typo.